auto_compressor/src/lib.rs

```diff
@@ -6,5 +6,47 @@
 //! to the database and uses these to enable it to incrementally work
 //! on space reductions
 
+use std::str::FromStr;
+
+use anyhow::Result;
+use synapse_compress_state::Level;
+
 pub mod manager;
 pub mod state_saving;
+
+/// Helper struct for parsing the `default_levels` argument.
+///
+/// The compressor keeps track of a number of Levels, each of which have a maximum length,
+/// current length, and an optional current head (None if level is empty, Some if a head
+/// exists).
+///
+/// This is needed since FromStr cannot be implemented for structs
+/// that aren't defined in this scope
+#[derive(PartialEq, Debug)]
+pub struct LevelInfo(pub Vec<Level>);
```

Comment:
Don't we already have a `LevelInfo` in the `synapse_compress_state` crate?

Reply:
Nope! `continue_run` has an argument called `level_info` which has type `&[Level]` - maybe that's what you were thinking of?

Reply:
Ah yeah, nevermind

Reply:
There is a `LevelSizes` struct that goes from comma seperated numbers to a vector of numbers though (which I guess I could reuse, but this way seems tidier)